frame/staking/src/benchmarking.rs

```diff
@@ -0,0 +1,703 @@
+use super::*;
+use sp_runtime::{BenchmarkResults, BenchmarkParameter, DispatchError};
+use sp_runtime::traits::{Dispatchable, Convert, Benchmarking, BenchmarkingSetup};
+use sp_io::hashing::blake2_256;
+use frame_support::StorageValue;
+use pallet_indices::address::Address;
+use frame_system::RawOrigin;
+use sp_phragmen::{
+	ExtendedBalance, StakedAssignment, reduce, build_support_map, evaluate_support, PhragmenScore,
+};
+use rand::Rng;
+use rand_chacha::rand_core::SeedableRng;
+
+macro_rules! assert_ok {
+	( $x:expr $(,)? ) => {
+		assert_eq!($x, Ok(()));
+	};
+	( $x:expr, $y:expr $(,)? ) => {
+		assert_eq!($x, Ok($y));
+	}
+}
+
+const CTRL_PREFIX: u32 = 1000;
+const NOMINATOR_PREFIX: u32 = 1_000_000;
+const USER: u32 = 999_999_999;
+
+static mut SEED: u64 = 999_666;
+static mut MODE: BenchmarkingMode = BenchmarkingMode::StrongerSubmission;
+
+type AddressOf<T> = Address<<T as frame_system::Trait>::AccountId, u32>;
+
+/// _r_andom number in _r_ange.
+fn rr(a: u32, b: u32) -> u32 {
+	// well, what do you expect? This is just benchmark code.
+	unsafe {
```

Comment:
dumb question, what about this is unsafe?

Reply:
Mutating a global static requires unsafe.